bridges/FurAffinityBridge.php

```diff
@@ -0,0 +1,914 @@
+				'defaultValue' => 10,
+				'title' => 'Limit number of submissions to return. -1 for unlimited.'
+			),
+			'full' => array(
+				'name' => 'Full view',
+				'title' => 'Include description, tags, date and larger image in article. Uses more bandwidth.',
+				'type' => 'checkbox',
+				'defaultValue' => 'checked'
+			),
+			'cache' => array(
+				'name' => 'Cache submission pages',
+				'title' => 'Reduces requests to FA when Full view is enabled. Changes to submission details may be delayed.',
+				'type' => 'checkbox',
+				'defaultValue' => 'checked'
+			)
+		),
+		'Gallery Folder' => array(
+			'username-folder' => array(
+				'name' => 'Username',
+				'required' => true,
+				'title' => 'Lowercase username as seen in URLs'
+			),
+			'folder-id' => array(
+				'name' => 'Folder ID',
+				'required' => true,
+				'type' => 'number',
+				'title' => 'Number seen in folder URL'
+			),
+			'limit' => array(
+				'name' => 'Limit',
+				'type' => 'number',
+				'defaultValue' => 10,
+				'title' => 'Limit number of submissions to return. -1 for unlimited.'
+			),
+			'full' => array(
+				'name' => 'Full view',
+				'title' => 'Include description, tags, date and larger image in article. Uses more bandwidth.',
+				'type' => 'checkbox',
+				'defaultValue' => 'checked'
+			),
+			'cache' => array(
+				'name' => 'Cache submission pages',
+				'title' => 'Reduces requests to FA when Full view is enabled. Changes to submission details may be delayed.',
+				'type' => 'checkbox',
+				'defaultValue' => 'checked'
+			)
+
+
+		)
+	);
+
+	/*
+	 * This was aquired by creating a new user on FA then
+	 * extracting the cookie from the browsers dev console.
+	 */
+	const FA_AUTH_COOKIE = 'b=4ce65691-b50f-4742-a990-bf28d6de16ee; a=ca6e4566-9d81-4263-9444-653b142e35f8';
+
+	public function detectParameters($url) {
+		$params = array();
+
+		// Single journal
+		$regex = '/^(https?:\/\/)?(www\.)?furaffinity.net\/journal\/(\d+)/';
+		if(preg_match($regex, $url, $matches) > 0) {
+			$params['journal-id'] = urldecode($matches[3]);
+			return $params;
+		}
+
+		// Journals
+		$regex = '/^(https?:\/\/)?(www\.)?furaffinity.net\/journals\/([^\/&?\n]+)/';
+		if(preg_match($regex, $url, $matches) > 0) {
+			$params['username-journals'] = urldecode($matches[3]);
+			return $params;
+		}
+
+		// Gallery folder
+		$regex = '/^(https?:\/\/)?(www\.)?furaffinity.net\/gallery\/([^\/&?\n]+)\/folder\/(\d+)/';
+		if(preg_match($regex, $url, $matches) > 0) {
+			$params['username-folder'] = urldecode($matches[3]);
+			$params['folder-id'] = urldecode($matches[4]);
+			$params['full'] = 'on';
+			return $params;
+		}
+
+		// Gallery (must be after gallery folder)
+		$regex = '/^(https?:\/\/)?(www\.)?furaffinity.net\/(gallery|scraps|favorites)\/([^\/&?\n]+)/';
+		if(preg_match($regex, $url, $matches) > 0) {
+			$params['username-' . $matches[3]] = urldecode($matches[4]);
+			$params['full'] = 'on';
+			return $params;
+		}
+
+		return null;
+	}
+
+	public function getName() {
```

Comment:
My OCD tells me that it becomes more readable if the cases look similar (look at getDescription for example).

* Put the `return` line on a separate line
* Concatenate on separate lines

```php
		switch($this->queriedContext) {
		case 'Search':
			return 'Search For '
				. $this->getInput('q');
		case 'Browse':
			return 'Browse';
		case 'Journals':
			return $this->getInput('username-journals');
		case 'Single Journal':
			return 'Journal '
				. $this->getInput('journal-id');
		case 'Gallery':
			return $this->getInput('username-gallery');
		case 'Scraps':
			return $this->getInput('username-scraps');
		case 'Favorites':
			return $this->getInput('username-favorites');
		case 'Gallery Folder':
			return $this->getInput('username-folder')
				. '\'s Folder '
				. $this->getInput('folder-id');
		default: return parent::getName();
```

Reply:
Yeah that looks better, added in dfba7a8, thanks!